perception/image_projection_based_fusion/src/roi_pointcloud_fusion/node.cpp

```diff
@@ -143,16 +155,37 @@ void RoiPointCloudFusionNode::fuseOnSingleImage(
         check_roi.x_offset <= normalized_projected_point.x() &&
         check_roi.y_offset <= normalized_projected_point.y() &&
         check_roi.x_offset + check_roi.width >= normalized_projected_point.x() &&
-        check_roi.y_offset + check_roi.height >= normalized_projected_point.y()) {
-        cluster.push_back(pcl::PointXYZ(*iter_orig_x, *iter_orig_y, *iter_orig_z));
+        check_roi.y_offset + check_roi.height >= normalized_projected_point.y() &&
+        clusters_data_size.at(i) < static_cast<size_t>(max_cluster_size_ * point_step)) {
+        std::memcpy(
+          &cluster.data[clusters_data_size.at(i)], &input_pointcloud_msg.data[offset], point_step);
+        clusters_data_size.at(i) += point_step;
       }
     }
   }
 
-  // refine and update output_fused_objects_
-  updateOutputFusedObjects(
-    output_objs, clusters, input_pointcloud_msg.header, input_roi_msg.header, tf_buffer_,
-    min_cluster_size_, cluster_2d_tolerance_, output_fused_objects_);
+  for (size_t i = 0; i < output_objs.size(); ++i) {
+    auto & feature_obj = output_objs.at(i);
+    auto & cluster = clusters.at(i);
+    if (
+      clusters_data_size.at(i) < static_cast<size_t>(min_cluster_size_ * point_step) ||
+      clusters_data_size.at(i) >= static_cast<size_t>(max_cluster_size_ * point_step)) {
```

Comment:
It's a minor but minor correction, but even the logic has been changed. Maybe you should change the title or separate the PR.

Reply:
@yukkysaito thank you for your comment, I think I will separate this PR to:
- simplify node 
- add intensity

Reply:
@yukkysaito I keep the same algorithm as before and just change to copy input pointcloud fields to output. 

Since this node mainly focus on small and less points unknown (traffic cones) objects detection which missed by Euclidean Clusters, so I would like to add `max_cluster_size` to limit the range of objects that roi_pointcloud fusion node processes. 
Could you review this again.

Reply:
@yukkysaito I understood. I split the parameter add to this PR https://github.com/autowarefoundation/autoware.universe/pull/6860
Could you review it too?

Reply:
:+1:

Reply:
> @badai-nguyen Adding max_cluster_size would be a logic change, so can we please change the title of the PR or split it?

@yukkysaito I made this PR only for keeping output pointcloud field as input topic. Could you review this again?